src/Components/Common/TemperatureFormField.tsx

```diff
@@ -1,77 +1,100 @@
-import { useState } from "react";
+import { useState, useEffect } from "react";
 import { FormFieldBaseProps } from "../Form/FormFields/Utils";
-import RangeAutocompleteFormField from "../Form/FormFields/RangeAutocompleteFormField";
+import { fahrenheitToCelsius, celsiusToFahrenheit } from "@/Utils/utils";
 import CareIcon from "../../CAREUI/icons/CareIcon";
 import ButtonV2 from "./components/ButtonV2";
-import { fahrenheitToCelsius } from "../../Utils/utils";
+import TextFormField from "../Form/FormFields/TextFormField";
 
 type TemperatureUnit = "celsius" | "fahrenheit";
 
-type Props = FormFieldBaseProps<number> & {
-  placeholder?: string;
-};
+type TemperatureFormFieldProps = FormFieldBaseProps<string>;
 
-export default function TemperatureFormField(props: Props) {
+export default function TemperatureFormField({
+  onChange,
+  id,
+  label,
+  error,
+  value,
+  name,
+}: TemperatureFormFieldProps) {
   const [unit, setUnit] = useState<TemperatureUnit>("fahrenheit");
+  const [inputValue, setInputValue] = useState(value || "");
+
+  useEffect(() => {
+    if (value) {
+      const initialTemperature =
+        unit === "celsius"
+          ? fahrenheitToCelsius(parseFloat(value)).toFixed(1)
+          : value;
+      setInputValue(initialTemperature);
+    }
+  }, [value, unit]);
+
+  const handleUnitChange = () => {
+    setUnit(unit === "celsius" ? "fahrenheit" : "celsius");
+    if (inputValue) {
+      const convertedValue =
+        unit === "celsius"
+          ? celsiusToFahrenheit(parseFloat(inputValue)).toFixed(1)
+          : fahrenheitToCelsius(parseFloat(inputValue)).toFixed(1);
+      setInputValue(convertedValue);
+    }
+  };
+
+  const handleInputChange = (e: any) => {
+    const newValue = e.value;
+
+    const regex = /^-?\d*\.?\d{0,1}$/;
+    if (regex.test(newValue)) {
+      setInputValue(newValue);
+    }
+  };
```

Comment:
my bad for not pointing this out earlier. `any` usages are not allowed when the type can be determined easily.

Reply:
No problem

Reply:
Done